include/libcyphal/platform/single_threaded_executor.hpp

```diff
@@ -113,114 +99,175 @@ class SingleThreadedExecutor : public IExecutor  // NOSONAR cpp:S4963
+
+            return *this;
+        }
+
+        CallbackNode(const CallbackNode&)            = delete;
+        CallbackNode& operator=(const CallbackNode&) = delete;
+
+        explicit operator Callback::Handle() const noexcept
+        {
+            // Next nolint & NOSONAR are unavoidable: this is opaque handle conversion code.
+            // NOLINTNEXTLINE(cppcoreguidelines-pro-type-reinterpret-cast)
+            return reinterpret_cast<Callback::Handle>(this);  // NOSONAR cpp:S3630
+        }
+        static CallbackNode* fromHandle(const Callback::Handle handle) noexcept
+        {
+            CETL_DEBUG_ASSERT(handle, "");
+
+            // Next nolint & NOSONAR are unavoidable: this is opaque handle conversion code.
+            // NOLINTNEXTLINE(cppcoreguidelines-pro-type-reinterpret-cast, performance-no-int-to-ptr)
+            return reinterpret_cast<CallbackNode*>(handle);  // NOSONAR cpp:S3630
+        }
+
+        SingleThreadedExecutor& executor() const noexcept
+        {
+            return executor_.get();
+        }
+
+        const Callback::Schedule::Variant& getSchedule() const noexcept
         {
             return schedule_;
         }
 
-        TimePoint& executionTime() noexcept
+        void setSchedule(const Callback::Schedule::Variant& schedule)
         {
-            return exec_time_;
+            schedule_ = schedule;
+            cetl::visit(cetl::make_overloaded(
+                            [this](const Callback::Schedule::None&) {
+                                //
+                                next_exec_time_ = TimePoint::max();
+                            },
+                            [this](const Callback::Schedule::Once& once) {
+                                //
+                                next_exec_time_ = once.exec_time;
+                            },
+                            [this](const Callback::Schedule::Repeat& repeat) {
+                                //
+                                next_exec_time_ = repeat.exec_time;
+                            }),
+                        schedule);
+        }
+
+        void reschedule(const TimePoint exec_time)
+        {
+            cetl::visit(cetl::make_overloaded(
+                            [this](const Callback::Schedule::None&) {
+                                //
+                                next_exec_time_ = TimePoint::max();
+                            },
+                            [this](const Callback::Schedule::Once&) {
+                                //
+                                next_exec_time_ = TimePoint::max();
+                            },
+                            [this, exec_time](const Callback::Schedule::Repeat& repeat) {
+                                //
+                                next_exec_time_ = exec_time + repeat.period;
+                            }),
+                        schedule_);
```

Comment:
This is just a silly nit, feel free to ignore, but

```suggestion            next_exec_time_ =
                cetl::visit(cetl::make_overloaded([](const Callback::Schedule::None&) { return TimePoint::max(); },
                                                  [](const Callback::Schedule::Once&) { return TimePoint::max(); },
                                                  [exec_time](const Callback::Schedule::Repeat& repeat) {
                                                      return exec_time + repeat.period;
                                                  }),
                            schedule_);
```